### ISSUE TITLE
Colour scheme (light/dark mode) does not automatically update according to changes in Mac's appearance settings (under General)

### ISSUE THREAD

**Issue by liweiyap** [liweiyap/text-popover-macOS#3]
I've tried adding [`.environment()`](https://stackoverflow.com/questions/62525731/unable-to-change-the-dark-mode-on-the-swiftui) and [`.colorScheme()`](https://stackoverflow.com/questions/61912363/swiftui-how-to-implement-dark-mode-toggle-and-refresh-all-views), the latter of which requiring a `@State` variable of type `ColorScheme`. However, there were two issues:

1. The same problem was still not solved.
2. It's just too simplistic to use `.environment()` and `.colorScheme()`. Only the background colour will change, but the text colour stays the same. We need to find a way to call `ContentView.toggleBackgroundColour()` when the `@Environment` variable of type `ColorScheme` gets updated from outside the application. We could, of course, write more code to change the make sure that the text colour is changed in tandem with the background colour, but there has to be a simpler solution.

I believe the correct solution is to use [`.onChange()`](https://developer.apple.com/documentation/swiftui/view/onchange(of:perform:)) in the `body` of [`ContentView`](https://github.com/liweiyap/text-popover-macOS/blob/master/text-popover-macOS/View/ContentView.swift). Unfortunately, this function is not currently available; it should hopefully be available in macOS version `11.0`. (Currently, at the time of writing, we're at version `10.15.6`.)

**Comment by liweiyap** [comment]
Update: in [`ContentView`](https://github.com/liweiyap/text-popover-macOS/blob/master/text-popover-macOS/View/ContentView.swift), I tried:

```
@ViewBuilder
var body: some View
{
  if #available(OSX 11.0, *)
  {
    HStack
    {
      // ...
    }
    .onChange(of: colorScheme, perform: {
      _ in
      
      toggleBackgroundColour()  // or: backgroundOptions.darkMode = !backgroundOptions.darkMode
    })
    // ...
  }
}
```

but not only did automatic detection still not work, but upon manual attempts at toggling in the settings window, I got the following error:

```
onChange(of: ColorScheme) action tried to update multiple times per frame.
```

**Comment by liweiyap** [comment]
The closest I've gotten is by following the solutions in this [link](https://stackoverflow.com/questions/39048894/how-to-detect-switch-between-macos-default-dark-mode-using-swift-3) and adding:
```
DistributedNotificationCenter.default.addObserver(self, selector: #selector(interfaceModeChanged(sender:)), name: NSNotification.Name(rawValue: "AppleInterfaceThemeChangedNotification"), object: nil)

func interfaceModeChanged(sender: NSNotification) {
  backgroundOptions.darkMode = (colorScheme == .dark)
}
```
to `AppDelegate`. This works when going from dark mode to light mode in the Mac's system preferences, but not from light mode to dark mode, whilst manual control with the settings window still works both ways. There must be some bug in my code but I haven't been able to find it yet.

**Comment by liweiyap** [comment]
Fixed in commit: 569e17303a8446b32bb4b994a1d9e22cbee3de09

The correct solution is:
```
DistributedNotificationCenter.default.addObserver(self, selector: #selector(interfaceModeChanged(sender:)), name: NSNotification.Name(rawValue: "AppleInterfaceThemeChangedNotification"), object: nil)

func interfaceModeChanged(sender: NSNotification) {
  backgroundOptions.darkMode = !backgroundOptions.darkMode
}
```

We cannot use the `colorScheme` variable because it is initialised to `@Environment(\.colorScheme)` at the start and then never changes again. When the colour scheme in the Mac's appearance settings changes, it is `@Environment(\.colorScheme)` that gets updated, not the local `colorScheme` variable.
